code/controllers/TaxonomyDirectoryController.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/**
+ * Class TaxonomyDirectoryController
+ *
+ * Controller for returning a list of pages tagged with a specific Taxonomy Term
+ */
+class TaxonomyDirectoryController extends Page_Controller
+{
+
+    private static $allowed_actions = array(
+        'index'
+    );
+
+    public function index(SS_HTTPRequest $HTTPRequest)
+    {
+        //look up the term to avoid sql injection by using the ORM
+        $term = TaxonomyTerm::get()->filter(['URLSegment' => $HTTPRequest->param('ID')])->first();
+
+        if ($term) {
+
+            $title = $term->Name;
+
+            $results = Page::get()
+                ->innerJoin(
+                    'BasePage_Terms',
```

Comment:
If these are proper relations on the model you should be able to do this all with the ORM's `filter`

Reply:
This was taken from the documentation in the developer.md in this module. Happy to revisit and we'd need to perhaps update the docs too.

Reply:
ok - if you don't mind adding that, then sure - otherwise you could leave it.